cai2hcl/cai2hcl.go

```diff
@@ -58,6 +64,37 @@ func Convert(assets []*caiasset.Asset, options *Options) ([]byte, error) {
 	return t, err
 }
 
+func groupAssets(assets []*caiasset.Asset) map[string][]*caiasset.Asset {
+	groups := make(map[string][]*caiasset.Asset)
+	var exceptions []*caiasset.Asset
+	for _, asset := range assets {
+		if asset == nil || asset.Type == "" || asset.Name == "" {
+			continue
+		}
+		if asset.Type == "bigtableadmin.googleapis.com/Cluster" || asset.Type == "cloudbilling.googleapis.com/ProjectBillingInfo" {
+			exceptions = append(exceptions, asset)
+		} else {
+			groups[asset.Name] = []*caiasset.Asset{asset}
+		}
+	}
+	for _, asset := range exceptions {
+		switch asset.Type {
+		case "cloudbilling.googleapis.com/ProjectBillingInfo":
```

Comment:
I'm a little concerned about hard-coding behavior related to a particular asset Type outside of the converters for that asset type; it seems like it might get hard to understand how a particular asset is supposed to behave. Would it be possible to have this logic be part of the converter?

Reply:
The converter itself does not keep a state. 
We need to group assets together before feeding them as parameter to converter.

Reply:
gotcha - so it would be possible to handle this logic in the converter, but it would require making the converters track an internal state and handle merging together the final resources for each asset. That would be basically the same way the converter tracks state for TF -> CAI conversion in TFV today.

What are the benefits of handling it this way (groupAssets) rather than having the converter track state?

Reply:
I don't get how TF-> CAI conversion is tracking state. Do you mean the ancestry manager cache? 
For example, there are billinginfo asset, and a project asset, the billinginfo just converts to a project name, and a billing string, and it stays in some cache? then when the converter sees the project asset, it checks the cache to see whether a billinginfo exist first? Then until we parse all the assets, we generate the hcl at that point.

Reply:
by cache I mean similar to https://github.com/GoogleCloudPlatform/terraform-validator/blob/cb5e546c31f8b52d22742d781545e72af71fdf94/converters/google/convert.go#L159

But really I don't care how this is implemented as long as the logic for whether something is grouped or not ultimately lies with the converter for that resource.